src/ExpandOnline/KlipfolioApi/Object/AbstractCrudObject.php

```diff
@@ -0,0 +1,163 @@
+<?php
+
+namespace ExpandOnline\KlipfolioApi\Object;
+
+use ExpandOnline\KlipfolioApi\Client;
+use ExpandOnline\KlipfolioApi\Request;
+
+/**
+ * Class AbstractCrudObject
+ * @package ExpandOnline\KlipfolioApi\Object
+ */
+abstract class AbstractCrudObject extends AbstractObject
+{
+
+    const FIELD_ID = 'id';
+
+    /**
+     * @var Client
+     */
+    private $client;
+
+    /**
+     * AbstractCrudObject constructor.
+     * @param null $id
+     * @param Client|null $client
+     */
+    public function __construct($id = null, Client $client = null)
```

Comment:
This means that applications using the API definetely need to use a factory to create CRUDObjects.